File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left=0;
        int right=nums.size()-1;
         int ans=0;

         while(left<=right){
            int mid=left+(right-left)/2;

            if(nums[mid]<=nums[nums.size()-1]){
                ans=nums[mid];
                right=mid-1;
            }
            else{
                left=mid+1;
            }
         }
         return ans;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution(object):
    def findMin(self, nums):
        left=0
        right=len(nums)-1
        ans=0
        while left<=right:
            mid=left+(right-left)//2
            if nums[mid]<= nums[len(nums)-1]:
                ans=nums[mid]
                right=mid-1
            else :
                left=mid+1
            
        return ans
        